engine/table/src/main/java/io/deephaven/engine/table/impl/TimeTable.java

```diff
@@ -33,37 +47,47 @@
 
     private static final String TIMESTAMP = "Timestamp";
     private long lastIndex = -1;
-    private final DateTimeArraySource dateTimeArraySource;
+    private final SyntheticDateTimeSource columnSource;
     private final TimeProvider timeProvider;
-    private DateTime lastTime;
-    private final long period;
-    private final long binOffset;
     private final PerformanceEntry entry;
+    private final boolean isStreaming;
 
     public TimeTable(TimeProvider timeProvider, long period) {
-        this(timeProvider, null, period);
+        this(timeProvider, null, period, false);
+    }
+
+    public TimeTable(TimeProvider timeProvider, long period, boolean isStreaming) {
+        this(timeProvider, null, period, isStreaming);
     }
 
     public TimeTable(TimeProvider timeProvider, DateTime firstTime, long period) {
-        super(RowSetFactory.fromKeys().toTracking(), initColumn());
+        this(timeProvider, firstTime, period, false);
+    }
+
+    public TimeTable(TimeProvider timeProvider, DateTime firstTime, long period, boolean isStreaming) {
+        super(RowSetFactory.fromKeys().toTracking(), initColumn(period));
```

Comment:
I know you didn't change this, but `empty()` would be more idiomatic than `fromKeys()`.